File: backend/src/main/java/com/prefin/domain/user/Parent.java
package com.prefin.domain.user;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import java.math.BigDecimal;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
@Table(name = "parents")
public class Parent {

    @Id
    @GeneratedValue
    private Long id;

    private String userId;

    private String password;

    private String name;

    private String fcmToken;

    private String account;

    private BigDecimal loanRate;

    private BigDecimal savingRate;

    private String simplePass;

}

File: mobile/Prefin/app/src/main/java/com/prefin/model/dto/Parent.kt
package com.prefin.model.dto

import java.math.BigDecimal

data class Parent(
    var account: String? = "",
    var balance: Int = 0,
    var fcmToken: String? = "",
    var id: Long = 0,
    var loanRate: BigDecimal? = BigDecimal("0.0"),
    var name: String = "",
    var password: String = "",
    var savingRate: BigDecimal? = BigDecimal("0.0"),
    var simplePass: String? = "",
    var userId: String = "",
    var maxSavingAmount: Int = 0,
) {
    constructor(): this(null, 0, null, 0, null, "", "", null, null, "")
}